plugins/auth-backend-module-bitbucket-server-provider/src/helpers.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fetch from 'node-fetch';
+import { PassportProfile } from '@backstage/plugin-auth-node';
+
+export async function fetchProfile(options: {
+  host: string;
+  accessToken: string;
+}): Promise<PassportProfile> {
+  const { host, accessToken } = options;
```

Comment:
Note that Bitbucket servers allow for the configuration of custom Api Base URLs. 
This means we cannot assume that the server address can be replicated using only a host address. Instead, we would need the entire `apiBaseUrl`. 

For example, a Bitbucket server address like "https://bitbucket.company-domain.com" is possible, but it's also possible to have a server address like "https://company-domain.com/bitbucket." 

However, fewer companies employ the latter strategy, with [Atlassian dropping support for Bitbucket Server](https://bitbucket.org/blog/cloud-migration-benefits). My organization has even opted to migrate to a different service provider rather than continue to use Bitbucket Server, with its caveats of custom API Base URLs.

Reply:
Hi @dhruvaldarji, thanks a lot for your feedback, I really appreciate that 🙏🏻. Just to confirm, you are suggesting to rename the configuration to `baseUrl`, right? Since in this pull request I was just porting the legacy provider code to the new auth module format, I wonder if it would be a good idea to make the changes you suggested in a follow-up pull request. Let's ask @freben what he suggests for us 🙂

Reply:
No changes requested. I'm sure this will be fine. The user base is dwindling anyway, since Atlassian dropped support.